### DISCUSSION TITLE
Advanced Logging Idea

### DISCUSSION THREAD

**Discussion by zion830** [euphony-io/euphony#204] [Ideas]
## Summary
When we use Euphony without RECORD_AUDIO permission, the process is killed with `IllegalStateException`

**Full error message**
```
java.lang.IllegalStateException: startRecording() called on an uninitialized AudioRecord
```
When I first saw this, it's hard to know at once that it is caused by a permission.
Even though it's already written in README, it's a pretty common mistake.
So I suggest this idea!

## How to improve
1. If listen() is called without RECORD_AUDIO permission, listen() function ends immediately to prevent death of the app
2. Print a log requiring the permission with `Log.e`
 
**Example**
```
Before calling listen(), you should acquire RECORD_AUDIO permission.
```

**Comment by designe** [comment]
Hi @zion830 !
these are great ideas 👍 I really like your suggestions.
Through RxManager's error handling improvement & notification by log, it'll be more convenient!

**Reply by designe** [reply]
Let me make an issue through `Create issue from discussion`!

**Reply by zion830** [reply]
Thanks a lot!!
